### PR TITLE
Add a test for InMemoryAssetReader.

### PR DESCRIPTION
**Pull request by matanlurey**
Proves that https://github.com/dart-lang/build/issues/241 is not a bug.

### CODE REVIEW

File: build_test/test/in_memory_reader_test.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    test('#findAssets should throw if rootPackage not supplied', () {
+      assetReader = new InMemoryAssetReader();
+      expect(
+        () => assetReader.findAssets(new Glob('lib/*.dart')),
+        throwsUnsupportedError,
+      );
+    });
+
+    test('#findAssets should files in lib/', () {
```
Comment (natebosch) [inline build_test/test/in_memory_reader_test.dart]: [nit] should find? should list?

File: build_test/test/in_memory_reader_test.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      expect(assetReader.findAssets(new Glob('lib/*.dart')), [libAsset]);
+    });
+
+    test('#findAssets should files in test/', () {
```
Comment (natebosch) [inline build_test/test/in_memory_reader_test.dart]: [nit] should find? should list?

Comment (matanlurey) [inline build_test/test/in_memory_reader_test.dart]: Done.

File: build_test/test/in_memory_reader_test.dart (hunk 1 above)
Comment (matanlurey) [inline build_test/test/in_memory_reader_test.dart]: Done.

**Pull request merged**